sdk/bootstrap/main.go

```diff
@@ -329,49 +329,37 @@ func isFile(path string) (bool, error) {
 	return false, nil
 }
 
-func GetLocalIP() string {
-    addrs, err := net.InterfaceAddrs()
-    if err != nil {
-        return ""
-    }
-    for _, address := range addrs {
-        // check the address type and if it is not a loopback the display it
-        if ipnet, ok := address.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
-            if ipnet.IP.To4() != nil {
-                return ipnet.IP.String()
-            }
-        }
-    }
-    return ""
+func GetLocalIP() (addr string, err error) {
+	ip, err := mesos.ContainerIP()
+
+	if err != nil {
+		return
+	}
+
+	addr = ip.String()
+	return
 }
+
 // main
 
 func main() {
 	args := parseArgs()
 
-    libprocess_ip, found := os.LookupEnv("LIBPROCESS_IP")
-
-    if !found {
-        log.Fatalf("Cannot find LIBPROCESS_IP")
-    }
+	libprocess_ip, err := GetLocalIP()
 
-    if libprocess_ip == "0.0.0.0" {
-        log.Printf("ILLEGAL you must be on the overlay network, getting you a new one!")
-        libprocess_ip = GetLocalIP()
-        if libprocess_ip == "" {
-            log.Fatalf("Failed to get new local IP")
-        }
+	if err != nil {
+		log.Fatalf("Cannot find the container's IP address: ", err)
+	}
 
-        err := os.Setenv("LIBPROCESS_IP", libprocess_ip)
-        if err != nil {
-            log.Fatalf("Failed to SET new LIBPROCESS_IP")
-        }
-    }
+	err = os.Setenv("LIBPROCESS_IP", libprocess_ip)
```

Comment:
Is this only going to apply to child processes of `bootstrap`? In practice bootstrap is run like this:

```
./bootstrap --foo && echo "other stuff here..."
```